webapp/src/components/AssessmentMetadataBar.tsx

```diff
@@ -0,0 +1,260 @@
+import React, { useState, useRef, useEffect } from 'react';
+
+import {
+  Avatar,
+  Divider,
+  List,
+  ListItem,
+  ListItemAvatar,
+  ListItemText,
+  Tooltip,
+  Typography,
+  Switch,
+} from '@mui/material';
+import InfoIcon from '@mui/icons-material/Info';
+import InventoryIcon from '@mui/icons-material/Inventory';
+import TimerIcon from '@mui/icons-material/Timer';
+import CalendarMonthIcon from '@mui/icons-material/CalendarMonth';
+import CheckCircleIcon from '@mui/icons-material/CheckCircle';
+import LightbulbIcon from '@mui/icons-material/Lightbulb';
+import { blue, grey } from '@mui/material/colors';
+
+import {
+  Assessment,
+  SubmissionStatus,
+  AssessmentType,
+  AssessmentSubmission,
+} from '../assets/data';
+import { formatDateTime } from '../helpers/dateTime';
+
+interface AssessmentMetadataBarProps {
+  assessment: Assessment;
+  submission: AssessmentSubmission;
+}
+
+const AssessmentMetadataBar = ({
+  assessment,
+  submission,
+}: AssessmentMetadataBarProps) => {
+  const dueTime = new Date(assessment?.dueDate!);
```

Comment:
It can be argued we care more about the endTime than the dueTime. So in this case, I'd switch it out from the current `dueTime` logic to something following the pattern:

const endTime =
- if (submission.submitted_at)
	- (submission.submitted_at)
- else if (assessment.time_limit && (submission.opened_at + assessment.time_limit < assessment.due_date))
	- (submission.opened_at + assessment.time_limit)
- else
	- assessment.due_date

... then not worry about displaying the due date, since the end time essentially becomes our due date anyway.